spring-cloud-gcp-docs/src/main/asciidoc/core.adoc

```diff
@@ -0,0 +1,110 @@
+== Spring Cloud GCP Core
+
+At the center of every Spring Cloud GCP module are the concepts of `GcpProjectIdProvider` and
+`CredentialsProvider`.
+
+Spring Cloud GCP provides a Spring Boot starter to auto-configure the core components.
+
+Maven coordinates, using Spring Cloud GCP BOM:
+
+[source,xml]
+----
+<dependency>
+    <groupId>org.springframework.cloud</groupId>
+    <artifactId>spring-cloud-gcp-starter-core</artifactId>
+</dependency>
+----
+
+Gradle coordinates:
+
+[source]
+----
+dependencies {
+    compile group: 'org.springframework.cloud', name: 'spring-cloud-gcp-starter-core', version: '1.0.0.BUILD-SNAPSHOT'
+}
+----
+
+=== Project ID
+
+`GcpProjectIdProvider` is a functional interface that returns a GCP project ID string.
+
+[source,java]
+----
+public interface GcpProjectIdProvider {
+	String getProjectId();
+}
+----
+
+The Spring Boot GCP Core starter auto-configures a `GcpProjectIdProvider`. If a
+`spring.cloud.gcp.project-id` property is specified, the provided `GcpProjectIdProvider` returns
+that property value.
+
+[source,java]
+----
+spring.cloud.gcp.project-id=my-gcp-project-id
+----
+
+Otherwise, the project ID is discovered based on a
+https://googlecloudplatform.github.io/google-cloud-java/0.20.1/apidocs/com/google/cloud/ServiceOptions.html#getDefaultProjectId--[set of rules]:
```

Comment:
Isn't there some "versionless" URL to use?
I'm do not worry about this from upgrade to upgrade

Reply:
Meant to get around this. Swapped 0.20.1 to latest.